packages/cashier/src/stores/payment-agent-store.ts

```diff
@@ -129,17 +143,13 @@ export default class PaymentAgentStore {
     }
 
     @action.bound
-    filterPaymentAgentList(bank) {
-        if (bank) {
-            this.filtered_list = [];
+    filterPaymentAgentList(bank?: number | string): void {
+        if (bank && typeof bank === 'string') {
+            this.filtered_list = [] as unknown as IObservableArray<TPartialPaymentAgentList>;
```

Comment:
refactor logic: 
if `bank` equal 0 (the alias for "All payment methods") -> we will return full payment_agent list
if `bank` equal 'some_string' (e.g 'visa') -> we will filter the list